### DISCUSSION TITLE
Plugin to integrate chalk

### DISCUSSION THREAD

**Discussion by rawnly** [remarkjs/remark#767] [Ideas]
Hello everyone, I need to display some markdown text in the terminal and I'd like to make it pretty. I was thinking that a plugin that transforms markdown into something like `chalk` syntax would be super cool.

https://github.com/chalk/chalk

**Comment by rawnly** [comment]
It should be dead simple, no support for headings, and may integrate with [`terminal-link`][1].
I'll start working on this this afternoon..

[1]: https://github.com/sindresorhus/terminal-link

**Reply by wooorm** [reply]
Do you meant to:
- display raw markdown, but syntax highlighted?
- Display rendered markdown, with ANSI colors?

**Reply by rawnly** [reply]
I'd like to render markdown with a basic support for `bold`, `strike`, `underline`, `italic`, and `links`. And **chalk** + **terminal-link** seems to fit this case.

**Reply by wooorm** [reply]
[`remark-man`](https://github.com/remarkjs/remark-man) might be of interest. It turns markdown into Roff. Roff is specifically used for man pages, but also used in many other things.
Maybe that already solves what you want to do, or otherwise it can serve as inspiration!

**Reply by rawnly** [reply]
Yes I saw `remark-man` but it isn't exactly what I was looking for 😅

**Reply by Trott** [reply]
> * Display rendered markdown, with ANSI colors?

@wooorm This is what I'm looking for. Do you have any recommendations? I'm trying to figure out a solution for https://github.com/nodejs/changelog-maker/issues/120. The ANSI color codes inserted by `chalk` are getting stripped out when we run the markdown through `remark-parse`  and `remark-stringify` for formatting here: https://github.com/nodejs/changelog-maker/blob/caeeabd3270eb07206b74656ded38a8d3573fe7d/format.mjs

**Reply by Trott** [reply]
Whoops, "display rendered markdown" is not quite what I'm looking for. I'm looking to format markdown without losing ANSI colors. (Should I open a new discussion/question?)

**Reply by wooorm** [reply]
It sounds like you mean that ANSI sequences are getting stripped. Could you check with https://spec.commonmark.org/dingus/ and GH on what’s supposed to happen, and reproduce with unified/remark/micromark to see who’se doing things differently?

**Reply by Trott** [reply]
Looking more, I wonder if my expectations are wrong and remark and everyone else is Doing It Right™.

**Reply by wooorm** [reply]
Yeah. I think so too. The bracket (`[`) in ASCI sequences *could* interfere with markdown.

```js
node -e 'console.log("\x1b[31mred](#)")'
```

Prints stuff in red. And in markdown:

[31mred](#)

^-- a link.
